mod-users/src/main/resources/domain/mod-users/features/users.feature

```diff
@@ -0,0 +1,151 @@
+Feature: Users tests
+
+  Background:
+    * url baseUrl
+    * callonce login testUser
+    * configure headers = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*'  }
+    * def status = true
+    * def lastName = call random_string
+    * def firstName = call random_string
+    * def username = call random_string
+    * def email = 'abc@pqr.com'
+
+  Scenario: Create a new User with PatronGroup.
+    * def username = call random_string
+    * def barcode = call random_numbers
+    * def uuid = call uuid1
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+
+  Scenario: Search user by barcode.
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { barcode: 2222}
+    * def uuid = call uuid1
+    * def username = call random_string
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { barcode: 3333}
+
+    Given path 'users?query=(barcode=2222)'
+    When method GET
+    Then status 200
+    And match response.users[0].barcode == '2222'
```

Comment:
This only checks that the first user found has the expected barcode. It doesn't seem to check that no other users are found, meaning this would pass if the second user was also found.

Maybe a check on the count of records found could avoid this.